src/autoschedulers/common/cmdline.h

```diff
@@ -779,7 +779,7 @@ class parser {
         std::string nam;
         char snam;
         std::string desc;
-        bool has;
+        bool has = false;
```

Comment:
I also find it weird to read a list of members like this that vary between initialized and not, depending on whether their initial value is a compile-time constant or not.

Reply:
This particular example isn't a good one for your point, IMHO (the strings were always initialized 'here' and not in the ctor, and `snam` was left uninitialized, period)

That said, I see your point here -- it is unfortunate that C++ has multiple ways to do this, often located in separate files, and so regardless of which convention we use here, we can't know for sure whether (and how) `snam` is initialized at all. 

I'm not married to this change, but I must admit that I do prefer the idea (and readability) of member vars being explicitly initialized at the declaration point -- it strikes me as putting more critical information about a members initial value along with its name and type. Unless you are violently allergic to this, though, why don't I leave it open here for a while and see if anyone else has thoughts pro or con. (And we can easily disable `modernize-use-default-member-init` entirely and revert those parts of the change, so if opinion is a "nay", I still think upgrading the tool versions is worthwhile)

(It would be interesting to imagine a world in which C++ required *all* variables to be explicitly initialized... with the ability to explicitly leave a field uninitialized via `= poison;` or something similar.)